ppml/kms-client/DecryptFile.py

```diff
@@ -0,0 +1,79 @@
+import requests
+import json
+import argparse
+import base64
+import time
+import random
+import hmac
+from hashlib import sha256
+import os
+from collections import OrderedDict
+import urllib.parse
+from cryptography.fernet import Fernet
+
+# default app auth conf
+appid = '202112101919'
+appkey = '202112345678'
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-ip', '--ip', type=str, help='ip address of the ehsm_kms_server', required=True)
+    parser.add_argument('-port', '--port', type=str, help='port of the ehsm_kms_server',default='3000', required=False)
+    parser.add_argument('-pkp', '--pkp', type=str, help='path of the primary key storage file', required=True)
+    parser.add_argument('-dkp', '--dkp', type=str, help='path of the data key storage file', required=True)
+    parser.add_argument('-efp', '--efp', type=str, help='path of the encrypted file to decrypt', required=True)
+    args = parser.parse_args()
+    ip = args.ip
+    port = args.port
+    pkp = args.pkp
+    dkp = args.dkp
+    efp = args.efp
+    return ip, port, pkp, dkp, efp
+
+def request_params(payload):
+    params = OrderedDict()
+
+    params["appid"] = appid
+    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
+    params["timestamp"] = int(time.time())
+
+    query_string = urllib.parse.urlencode(params)
+    query_string += '&app_key=' + appkey
+    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), query_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8')
+
+    params["sign"] = sign
+    params["payload"] = payload
+    return params
+
+def decrypt_data_key(ip, port, primary_key_path, data_key_path):

```

Comment:
better to have 3 modules
kms rest client, put basic rest funtions in it
key management call kms rest client
file operation call key management